packages/react-sdk/src/components/link.tsx

```diff
@@ -1,13 +1,57 @@
+import { Link as RemixLink, useLocation } from "@remix-run/react";
 import { forwardRef, type ElementRef, type ComponentProps } from "react";
 
+const isAbsoluteUrl = (href: string) => {
+  // http:, mailto:, tel:, etc.
+  return /^[a-z]+:/i.test(href);
+};
+
+// @todo: Ideally this code should live in designer app.
+// Sdk should not know how routing is implemented in designer.
+// We can move it there later when we have a way of customizing components per environment.
+const preserveBuildParams = (href: string, currentSearch: string) => {
+  if (isAbsoluteUrl(href)) {
+    return href;
+  }
+
+  const [path, search] = href.split("?");
```

Comment:
please don;t what if user will pass `/ddd?ddd/?d=1` are u sure you will parse it same as browser